packages/core/src/credential/Credential.ts

```diff
@@ -301,10 +301,15 @@ export function verifyWellFormed(
   verifyDataIntegrity(credential)
 
   if (ctype) {
+    if (`kilt:ctype:${credential.claim.cTypeHash}` !== ctype.$id) {
+      throw new SDKErrors.CTypeIdMismatchError(
+        ctype.$id,
+        credential.claim.cTypeHash
+      )
+    }
```

Comment:
I was wondering if we wanted to do it another way.

```
if (ctype && `kilt:ctype:${credential.claim.cTypeHash}` !== ctype.$id) {
      throw new SDKErrors.CTypeIdMismatchError(
        ctype.$id,
        credential.claim.cTypeHash
      )
    }
if (ctype) {
 verifyClaimAgainstSchema(credential.claim.contents, ctype)
  }
```

Reply:
no, this is good, but we've got the CType.hashToId() helper function for `kilt:ctype:${credential.claim.cTypeHash}`